liferay-content-management/liferay-portal
# LPD-26298 Custom date filter in Content Dashboard

Custom date filter in Content Dashboard

# What is this trying to solve?

[Link to ticket](https://issues.liferay.com/browse/LPD-26298)

The PR adds the option to sort the results in the content dashboard by specific date ranges 

# How am I fixing it?

Modify the logic that we currently have in content dashboard to be able to filter by a range of dates of a given date type

# How can you verify that it works?

The PR includes a set of unit tests

1. An unit test to check that when the correct parameters are set, they are taken into account when building the search context
2. An unit test to check that when a non existing date type is selected, no filters are applied
3. An unit test to check that when non valid dates are selected, no filters are applied

The PR can also be tested through the following steps

1. Add a new Web Content or image
2. Go to Content Dashboard
3. In the toolbar, select Custom Date 
4. In the popup, select "Create Date"
5. Select the range of the dates
6. Accept
7. If the range of dates includes the web content, it will be shown

This is nice to test with a portal that has content created throughout several dates, to see the real effect from the filter. The database can also be manipulated manually to change the dates

![filter](https://github.com/liferay-content-management/liferay-portal/assets/4267448/838fc5f6-4b92-462d-a6bb-f601d6e57909)


@sandrodw3
